### PR TITLE
cmake: remove toolchain version check

### PR DESCRIPTION
**Pull request by sylvioalves**
This removes cmake gcc, ctng and crosstool check.
All those tools are not esp-idf related so this
check and warning messages can be removed.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

### CODE REVIEW

File: components/esp32/project_include.cmake [hunk 1]
```diff
@@ -26,6 +26,7 @@ endif()
 # Warn if the toolchain version doesn't match
 #
 # TODO: make these platform-specific for diff toolchains
-get_expected_ctng_version(expected_toolchain expected_gcc)
-gcc_version_check("${expected_gcc}")
-crosstool_version_check("${expected_toolchain}")
+# Zephyr: remove ctng, gcc and version check
+# get_expected_ctng_version(expected_toolchain expected_gcc)
+# gcc_version_check("${expected_gcc}")
+# crosstool_version_check("${expected_toolchain}")
```
Comment (glaubermaroto) [inline components/esp32/project_include.cmake:32]: What is the need to keep the original lines commented?

File: tools/cmake/idf.cmake [hunk 2]
```diff
@@ -21,7 +21,8 @@ if(NOT __idf_env_set)
                             ${idf_path}")
         endif()
     else()
-        message(WARNING "IDF_PATH environment variable not found. Setting IDF_PATH to '${_idf_path}'.")
+        # Zephyr: remove IDF_PATH build warning
+        # message(WARNING "IDF_PATH environment variable not found. Setting IDF_PATH to '${_idf_path}'.")
```
Comment (glaubermaroto) [inline tools/cmake/idf.cmake:25]: Same question here.

File: components/esp32/project_include.cmake (hunk 1 above)
Comment (sylvioalves) [reply components/esp32/project_include.cmake:32]: The only reason to keep original content commented is to make easy during next hal update. What do you think?

**Comment by sylvioalves** [comment]
@uLipe PTAL.

File: components/esp32/project_include.cmake (hunk 1 above)
Comment (glaubermaroto) [reply components/esp32/project_include.cmake:32]: I see, it is here to ease rebasing. My concern is that it potentially messes up the codebase in the mid/long run.

**Review by glaubermaroto — APPROVED** [review]

**Review by uLipe — APPROVED** [review]

**Pull request merged**